contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -468,6 +468,34 @@ is_login(Oid role_oid)
 	return is_login;
 }
 
+bool
+is_active_login(Oid role_oid)
+{
+	bool	is_active_login;
+	int		num_backends = pgstat_fetch_stat_numbackends();
+	int		curr_backend;
+
+	/* 1-based index */
+	for (curr_backend = 1; curr_backend <= num_backends; curr_backend++)
+	{
+		/*
+		 * Fetch the userid for active backend from pg_statistics and check if
+		 * given role_oid matches with curr_backend's userid, if it matches then
+		 * it is active login.
+		 */
+		Oid	backend_userid;
+		backend_userid = DatumGetObjectId(
+									DirectFunctionCall1(pg_stat_get_backend_userid,
```

Comment:
1. Now instead of API call, we're fetching the backend entry(PgBackendStatus) which maintain by pg statistics collector.
2. Instead of checking for all the active backends, now only client backends are being considered.[Note: We get valid(not null) userid for client-backends, wal-sender and bgworker processes, other processes gives NULL userid]